### ISSUE TITLE
Trust Authentification is not possible with Ecto 

### ISSUE THREAD

**Issue by jpbecotte** [elixir-ecto/ecto#3212]
### Precheck

* Do not use the issues tracker for help or support requests (try Elixir Forum, Stack Overflow, IRC or mailing lists, etc).
* For proposing a new feature, please start a discussion on [elixir-ecto](https://groups.google.com/forum/#!forum/elixir-ecto).
* For bugs, do a quick search and make sure the bug has not yet been reported.
* Finally, be nice and have fun!

### Environment

* Elixir version (elixir -v): 1.9.4 (compiled with Erlang/OTP 21)
* Database and version (PostgreSQL 9.4, MongoDB 3.2, etc.): postgres (PostgreSQL) 10.10 (Ubuntu 10.10-0ubuntu0.18.04.1)
* Ecto version (mix deps): 3.2.5
* Database adapter and version (mix deps): postgrex 0.15.2
* Operating system: Ubuntu 18.04.3 LTS

### Current behavior

Using `postgres:///db` as DATABASE_URL gives this error: « invalid url postgres:///db, host is not present. ».

### Expected behavior

It looks like we can't use Trust Authentification with Ecto. It requires a host, a password and a user. We should be able to do so.

`psql postgres:///db` on the command line works as expected.

**Comment by wojtekmach** [comment]
There's a bug in URI.parse and there's a fix already on Elixir v1.10.0-rc.0. With the linked PR you should be good to go.

**Comment by jpbecotte** [comment]
Hi @wojtekmach ! Looks fantastic. Thanks, I will test once the PR is merged.

**Comment by jpbecotte** [comment]
@wojtekmach : the original error is gone. But I got this error now : exception: key :password not found. Should I open a new ticket?

```
** (RuntimeError) connect raised KeyError exception: key :password not found. The exception details are hidden, as they may contain sensitive data such as database credentials. You may set :show_sensitive_data_on_connection_error to true when starting your connection if you wish to see all of the details
    (elixir 1.10.0-rc.0) lib/keyword.ex:399: Keyword.fetch!/2
    (postgrex 0.15.2) lib/postgrex/protocol.ex:722: Postgrex.Protocol.auth_md5/4
    (postgrex 0.15.2) lib/postgrex/protocol.ex:579: Postgrex.Protocol.handshake/2
    (db_connection 2.1.1) lib/db_connection/connection.ex:69: DBConnection.Connection.connect/2
    (connection 1.0.4) lib/connection.ex:622: Connection.enter_connect/5
    (stdlib 3.11.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
```

**Comment by wojtekmach** [comment]
Are you sure you have the trust method in your pg_hba.conf? Because looking at the stacktrace,

> (postgrex 0.15.2) lib/postgrex/protocol.ex:722: Postgrex.Protocol.auth_md5/4

it looks like you're hitting the md5 authentication method.

**Comment by jpbecotte** [comment]
Yes. I works now I have used the right rules for the pg_hba.conf. Thank you.

**Comment by nacengineer** [comment]
For someone that comes along later on this one... 

Keep in mind your postgres might be setup to use sockets and that may be causing the error depending on how your firewall works. For me on centos with 3rd party postgres, so sockets are in /tmp not in /var/run/postgresql, I had to set socket_dir to be /tmp

**Comment by DavidOliver** [comment]
Another pointer for others looking to use trusted auth with socket connection:

I'm currently using trust authentication and the socket is in `/var/run/postgresql`, and I still need to set `socket_dir` to avoid Ecto using TCP/localhost.

```
  database_socket_dir = System.get_env("DATABASE_SOCKET_DIR") || "/var/run/postgresql"

  config :my_app, MyApp.Repo,
    url: database_url,
    socket_dir: database_socket_dir,
```
